### PR TITLE
add 438. Find All Anagrams in a String

### PR DESCRIPTION
**Pull request by dyljqq**

### CODE REVIEW

File: String/FindAllAnagramInString.swift [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      return anas
+    }
+    
+    var dp: [Character: Int] = [:]
+    for c in p {
+      dp[c] = (dp[c] ?? 0) + 1
+    }
+    
+    var d: [Character: Int] = [:]
+
```
[the remark this answers is not in the document]
Comment (dyljqq) [reply String/FindAllAnagramInString.swift]: please review it again.

File: String/FindAllAnagramInString.swift [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+class FindAllAnagramInString {
+  
+  func findAnagrams(_ s: String, _ p: String) -> [Int] {
+    var anas = [Int]()
+    
+    if s == "" || p == "" {
+      return anas
+    }
+    
+    var dp: [Character: Int] = [:]
```
[the remark this answers is not in the document]
Comment (jotshin) [reply String/FindAllAnagramInString.swift]: @soapyigu What would you think about [this guideline](https://github.com/raywenderlich/swift-style-guide#type-annotation-for-empty-arrays-and-dictionaries)?

File: String/FindAllAnagramInString.swift [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/**
+ * Question Link: https://leetcode.com/problems/find-all-anagrams-in-a-string/description/
+ * Primary idea: Slide window and use left and right point to maintain, and to compare p and substring s.
+ *
+ * Time Complexity: O(n), n stands for number of s
+ *                  Space Complexity: O(n)
```
Comment (soapyigu) [inline String/FindAllAnagramInString.swift]: Indention error: No space before "Space"

File: String/FindAllAnagramInString.swift [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+ extension Dictionary where Key == Character, Value == Int  {
+  func isSame(_ dictionary: [Character: Int]) -> Bool {
```
Comment (soapyigu) [inline String/FindAllAnagramInString.swift]: Indention

File: String/FindAllAnagramInString.swift [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      if dictionary[key] != value {
+        return false
+      }
+    }
+    return true
+  }
+}
+
+class FindAllAnagramInString {
+
```
Comment (soapyigu) [inline String/FindAllAnagramInString.swift]: Delete the line.

File: String/FindAllAnagramInString.swift [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    func findAnagrams(_ s: String, _ p: String) -> [Int] {
+      var anas = [Int]()
+
+      if s == "" || p == "" {
+        return anas
+      }
+
+      var dp = [Character: Int]()
```
Comment (soapyigu) [inline String/FindAllAnagramInString.swift]: Could you put all variables at the top of function?

File: String/FindAllAnagramInString.swift [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+      for c in p {
+        dp[c] = (dp[c] ?? 0) + 1
+      }
+
+      var d = [Character: Int]()
+
+      var (left, right) = (0, 0)
+
+      for c in s {
```
Comment (soapyigu) [inline String/FindAllAnagramInString.swift]: Please give another name for c. c is very ambiguous as a variable name.

File: String/FindAllAnagramInString.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (soapyigu) [reply String/FindAllAnagramInString.swift]: I see thanks for justification!

**Comment by soapyigu** [comment]
Hi @dyljqq, 

Sorry for the delay and I just reviewed it again. Overall LGTM, however there are some remaining issues unresolved.

Please take a look at my comments and once they are resolved, you could click "Resolve conversation" button to resolve it. 

Thanks again for your support and help!

**Comment by soapyigu** [comment]
I am going to close this one at the end of May 4 if @dyljqq does not update it.

**Comment by soapyigu** [comment]
Close this one as no response from contributor.

**Pull request closed without merging**